sessions/04-using-delay-distributions-to-model-the-data-generating-process-of-an-epidemic.qmd

```diff
@@ -3,3 +3,268 @@ title: "Using delay distributions to model the data generating process"
+
+$$
+C_t = \sum_{i=0}^{i=p_\mathrm{max}} I_{t-i} p_i
+$$
+
+In words, the number of observable outcomes on day $t$ is given by the sum of infections on all previous days multiplied by the probability that those infections are observed on day $t$.
+For example, the observable outcomes $C_t$ could be the number of symptom onsets on day $t$ and $p_i$ is the incubation period.
+
+We can use exactly the same data as in the [previous session](03-biases-in-delay-distributions), again assuming a gamma-distributed incubation period with shape 5 and rate 1 and this time working with symptom onsets:
+
+```{r onsets}
+data(infection_times)
+df <- infection_times |>
+  mutate(
+    onset_time = infection_time + rgamma(n(), shape = 5, rate = 1)
+  )
+```
+
+We then aggregate this into a daily time series of infections
+
+```{r aggregate}
+## convert to days
+df <- df |>
+  transmute(
+    infection_day = floor(infection_time),
+    onset_day = floor(onset_time)
+  )
+## infection time series
+inf_ts <- df |>
+  count(infection_day, name = "infections")
+head(inf_ts)
+```
+
+The resulting time series does contain any days with no infections.
+We can make our calculations easier (as otherwise we would have to try and detect these in any models that used this data which could be complicated) by adding them:
+
+```{r missing_to_zero}
+all_days <- expand(
+  df, infection_day = seq(min(infection_day), max(infection_day))
+)
+inf_ts <- all_days |>
+  full_join(inf_ts, by = join_by(infection_day)) |>
+  replace_na(list(infections = 0))
+head(inf_ts)
+```
+
+Now we can convolve the time series with a delay distribution of outcomes to get a time series of outcomes as suggested above.
+However, if we want to use the gamma distribution with shape 5 and rate 1 as before we face a familiar issue.
+The gamma distribution is a _continuous_ distribution but now we have data in days which are _discrete_ entities.
+In order to simulate a time series of symptom onsets, we first generate a discrete probability mass function of the incubation period (in days) by generating many replicates of the corresponding random delay, taking into account that we have already rounded down our infection times to infection days.
+
+In the absence of any other information or model, we assume for our simulation that infection occurred at some random time during the day, with each day equally likely.
+We then apply the incubation period before once again rounding down to get the day of symptom onset.
+We repeat this many times to get the probabilty mass function that allows us to go from infection days to symptom onset days
+
+```{r discretise_2_day_window, file = here::here("functions", "censored-delay-pmf.r")}
```

Comment:
```suggestion
```{r double-censored-delay, file = here::here("functions", "censored-delay-pmf.r")}
```

Reply:
This can be dealt with in #82